### PR TITLE
Never type is stable in function return position

### PR DESCRIPTION
**Pull request by Sky9x**
#2

### CODE REVIEW

**Comment by Sky9x** [comment]
oh wait did you want the trait method or free `unreachable_checked` fn

**Comment by djkoloski** [comment]
I'd prefer `unreachable_checked`. The trait method is fine but the free function can be called in const contexts.

File: src/lib.rs [hunk 1]
```diff
@@ -289,6 +289,9 @@ pub trait OptionExt<T> {
 /// It must be impossible to produce a value of this type.
 pub unsafe trait Never {}
 
+#[inline(always)]
+const fn unreachable_checked<T: Never>(_: T) -> ! { unsafe { unreachable_unchecked() } }
```
Comment (djkoloski) [inline src/lib.rs]:
```suggestion
pub const fn unreachable_checked<T: Never>(_: T) -> ! { unsafe { unreachable_unchecked() } }
```
`Never` for the people! 😄

**Review by djkoloski — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (Sky9x) [reply src/lib.rs]: what should the docs say

**Review by djkoloski — APPROVED** [review]

**Comment by Sky9x** [comment]
:tada:

**Comment by djkoloski** [comment]
Thanks for the PR! I've only released `-pre` releases of rancor so far since it's primarily for use with rkyv and bytecheck. I planned on releasing them all around the same time (i.e. rancor at `0.1.0`), but that might not be for a while. Were you interested in using it in the near future?

**Comment by Sky9x** [comment]
I'm using rkyv for a project, and I was just exploring the github org. This is a cool library, and I might use it in the future, but rn I just wanted to remove some unsafe code :smiley: :crab:

**Pull request merged**